deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/constraint/BaseConstraint.java

```diff
@@ -20,18 +22,18 @@
 
     protected boolean applyToWeights;
     protected boolean applyToBiases;
+    protected Set<String> paramNames = new HashSet<>();
```

Comment:
It should be IMO one or the other - applyToWeights/applyToBiases OR paramNames set - not both. My vote is for the latter only.